### PR TITLE
Source:Kafka - Avro format message support 

### PR DESCRIPTION
**Pull request by sivankumar86**
## What
*Describe what the change is solving*
    User would be able to read/sync avro format message from kafka queue 

## How
   I have added KafkaAvroSerializer API to deserialize the Avro message. Please refer 
https://docs.confluent.io/platform/current/schema-registry/serdes-develop/serdes-avro.html for more details. 

## 🚨 User Impact 🚨
   This would be helpful to many  users who are using avro format  and want to integrate with Aribyte

It would resovles #15640 #15641

### CODE REVIEW

**Comment by sivankumar86** [comment]
@marcosmarxm would it be possible to you to assist on this change review ? I would appreciate it.

**Comment by sajarin** [comment]
Assigned to @prasrvenkat

**Comment by sivankumar86** [comment]
@sajarin @prasrvenkat Thanks for checking the pull request. I would appreciate your feedback and marge to master asap.

**Comment by dariof28** [comment]
Cool! When will this be available on master?

**Comment by ccianciulli** [comment]
Waiting for this merge too... 👍

**Comment by ptkdev** [comment]
Any news on this merge?

**Comment by sajarin** [comment]
Sorry for the delay everyone, this PR is part of Airbyte's maintainer program and we're trying to have external maintainers review community PRs. We hope to get this merged into master by the end of this week. 

assigned to @koconder

**Comment by koconder** [comment]
Hi all, basic checks and intergration tests looks good. Just need to run a few more checks on my side before we can approve this PR

File: airbyte-integrations/connectors/source-kafka/build.gradle
Comment (koconder) [inline airbyte-integrations/connectors/source-kafka/build.gradle:16]: @sivankumar86 You need to update gradle dependencies as per `docs/contributing-to-airbyte/gradle-dependency-update.md` in the file `deps.toml` in the project root. An example is the package `com.amazonaws:aws-java-sdk-s3` which is in the toml and used by the `destination-s3` connector

File: airbyte-integrations/connectors/source-kafka/src/main/java/io/airbyte/integrations/source/kafka/KafkaFormatFactory.java
Comment (koconder) [inline airbyte-integrations/connectors/source-kafka/src/main/java/io/airbyte/integrations/source/kafka/KafkaFormatFactory.java:15]: We can later extend this out to include PROTOBUF support using confluent libraries - Solves #15059 and #15059

File: airbyte-integrations/connectors/source-kafka/src/main/java/io/airbyte/integrations/source/kafka/format/AbstractFormat.java
Comment (koconder) [inline airbyte-integrations/connectors/source-kafka/src/main/java/io/airbyte/integrations/source/kafka/format/AbstractFormat.java:32]: @sivankumar86 Lots of whitespace have you ran `./gradlew format`

File: airbyte-integrations/connectors/source-kafka/src/main/resources/spec.json
Comment (koconder) [inline airbyte-integrations/connectors/source-kafka/src/main/resources/spec.json:48]: I ran a local unit test to validate how empty strings are treated, as you are not sanitizing input and taking it directly in. This passed the test.

File: airbyte-integrations/connectors/source-kafka/src/main/resources/spec.json
Comment (koconder) [inline airbyte-integrations/connectors/source-kafka/src/main/resources/spec.json:275]: @sivankumar86 rewrite to `The Maximum to be processed per execution`

File: docs/integrations/sources/kafka.md
Comment (koconder) [inline docs/integrations/sources/kafka.md:51]: @sivankumar86 rename `Avro` to `AVRO`, also include in the notes `Support for maximum records to process`

File: docs/integrations/sources/kafka.md
Comment (koconder) [inline docs/integrations/sources/kafka.md:44]: @sivankumar86 rename `Json` to `JSON` and `Avro` to `AVRO`

**Review by vincentkoc — CHANGES_REQUESTED** [review]
@sivankumar86 all the unit testing and changes are all good, just some cosmetic change:

* Adding confluent maven packages to the main deps.toml
* Reword some markdown files and spec
* Update the rest of the docs in relation to AVRO, Schema Registry and Max Topic support on`docs/integrations/sources/kafka.md`

File: airbyte-integrations/connectors/source-kafka/src/main/resources/spec.json
Comment (koconder) [inline airbyte-integrations/connectors/source-kafka/src/main/resources/spec.json:42]: @sivankumar86 we need to add notes that Schema Registry is only supported on the AVRO format not JSON?

Comment (koconder) [reply airbyte-integrations/connectors/source-kafka/src/main/resources/spec.json:42]: Please add to `docs/integrations/sources/kafka.md`

File: airbyte-integrations/connectors/source-kafka/build.gradle
Comment (sivankumar86) [reply airbyte-integrations/connectors/source-kafka/build.gradle:16]: It seems, it is only for common library and above is specific to kafka. let me know if you still want to move to root because, it does not make sense.

File: airbyte-integrations/connectors/source-kafka/src/main/java/io/airbyte/integrations/source/kafka/format/AbstractFormat.java
Comment (sivankumar86) [reply airbyte-integrations/connectors/source-kafka/src/main/java/io/airbyte/integrations/source/kafka/format/AbstractFormat.java:32]: it seems, there is no task. Task 'format' not found in project ':airbyte-integrations:connectors:source-kafka'.

File: airbyte-integrations/connectors/source-kafka/src/main/resources/spec.json
Comment (sivankumar86) [reply airbyte-integrations/connectors/source-kafka/src/main/resources/spec.json:48]: empty string would be removed before adding to kafka client hence, it is ok.

File: airbyte-integrations/connectors/source-kafka/src/main/java/io/airbyte/integrations/source/kafka/format/AbstractFormat.java
Comment (sivankumar86) [reply airbyte-integrations/connectors/source-kafka/src/main/java/io/airbyte/integrations/source/kafka/format/AbstractFormat.java]: Done

**Comment by sivankumar86** [comment]
@koconder  thank  you for taking a look. Could you take a look on new changes ?

**Comment by koconder** [comment]
@sivankumar86 its all good, just waiting on a final confirmation from team if we need to add new maven dep to toml file and a final intergation test

**Review by grishick — APPROVED** [review]

**Comment by sivankumar86** [comment]
@koconder @sajarin @grishick Thank you for reviewing the changes. Could you merge to master  ?

**Review by vincentkoc — APPROVED** [review]

**Comment by sajarin** [comment]
Thanks @sivankumar86 for the PR and thank you @koconder for the review!

**Pull request merged**
